dmscripts/export_framework_applicant_details.py

```diff
@@ -0,0 +1,137 @@
+import sys
+if sys.version_info[0] < 3:
+    import unicodecsv as csv
+else:
+    import csv
+
+from multiprocessing.pool import ThreadPool
+
+
+LOTS = {
+    "g-cloud-8": ["saas", "paas", "iaas", "scs"]
+}
+
+DECLARATION_FIELDS = {
+    "g-cloud-8": [
+        "primaryContact",
+        "primaryContactEmail",
+        "nameOfOrganisation",
+        "registeredAddressBuilding",
+        "registeredAddressTown",
+        "registeredAddressPostcode",
+        "tradingStatus",
+        "tradingStatusOther",
+        "tradingNames",
+        "firstRegistered",
+        "currentRegisteredCountry",
+        "companyRegistrationNumber",
+        "dunsNumber",
+        "registeredVATNumber",
+        "establishedInTheUK",
+        "appropriateTradeRegisters",
+        "appropriateTradeRegistersNumber",
+        "licenceOrMemberRequired",
+        "licenceOrMemberRequiredDetails",
+        "organisationSize",
+        "subcontracting",
+        "contactNameContractNotice",
+        "contactEmailContractNotice",
+        "cyberEssentials",
+        "cyberEssentialsPlus",
+    ]
+}
+
+
+def find_suppliers(client, framework_slug):
+    suppliers = client.get_interested_suppliers(framework_slug)['interestedSuppliers']
+    return ({'supplier_id': supplier_id} for supplier_id in suppliers)
+
+
+def add_supplier_info(client):
+    def inner(record):
+        supplier = client.get_supplier(record['supplier_id'])
+
+        return dict(record,
+                    supplier=supplier['suppliers'])
+
+    return inner
+
+
+def add_framework_info(client, framework_slug):
+    def inner(record):
+        supplier_framework = client.get_supplier_framework_info(record['supplier_id'], framework_slug)
+        supplier_framework = supplier_framework['frameworkInterest']
+        supplier_framework['declaration'] = supplier_framework['declaration'] or {}
+
+        return dict(record,
+                    declaration=supplier_framework['declaration'],
+                    onFramework=supplier_framework['onFramework'])
+
+    return inner
+
+
+def add_submitted_draft_counts(client, framework_slug):
+    def inner(record):
+        counts = {lot: 0 for lot in LOTS[framework_slug]}
+
+        for draft in client.find_draft_services_iter(record['supplier']['id'], framework=framework_slug):
+            if draft['status'] == 'submitted':
```

Comment:
It's interesting that we don't have a `status` filter for `/drafts`. We should think about adding this at some point.